File: ProductCrudKnockOut/Models/ProductModel.cs
using System.ComponentModel.DataAnnotations;

namespace ProductCrudKnockOut.Models
{
    public class ProductModel
    {
        [Key]
        public int Id {  get; set; }

        [Display(Name ="Product Name")]
        public string Name { get; set; }

        [Display(Name = "Product Quantity")]
        public int Quantity {  get; set; }

        [Display(Name="Company Name")]
        public string Company { get; set; }
     
        [Display(Name="Manufactured Date")]
        public DateTime ManufacturedDate { get; set; }

        [Display(Name = "Expiry Date")]
        public DateTime ExpiryDate { get; set; }

        [Display(Name="Place Order")]
        public string Order { get; set; }


    }
}

File: ProductCrudKnockOut/wwwroot/js/Product/ProductModel.js
/// <reference path="../knockout.js" />

var productModel = function (item)
{
    var self = this;
    item = item || {};
    self.id = ko.observable(item.id || 0);
    self.name = ko.observable(item.name || '');
    self.quantity = ko.observable(item.quantity || 0);
    self.company = ko.observable(item.company || '');
    self.manufacturedDate = ko.observable(item.manufacturedDate || '');
    self.expiryDate = ko.observable(item.expiryDate || '');
    self.order = ko.observable(item.order || '');
}